code/game/objects/items/devices/laserpointer.dm

```diff
@@ -96,7 +96,7 @@
 				severity = 0
 
 			//chance to actually hit the eyes depends on internal component
-			if(prob(effectchance * diode.rating) && C.flash_act(severity))
+			if((C == user || prob(effectchance * diode.rating)) && C.flash_act(severity))
 				outmsg = "<span class='notice'>You blind [C] by shining [src] in their eyes.</span>"

```

Comment:
Can you make the message reflect the fact that you specifically blinded yourself and not another guy with your name

Reply:
make people with the same name have 100% blind chance instead :^)